src/hal/src/external_memory/fd.rs

```diff
@@ -0,0 +1,144 @@
+use super::{ExternalMemory, ExternalMemoryType, ExternalMemoryTypeFlags};
+
+/// Unix file descriptor
+#[derive(Debug)]
+pub struct Fd(i32);
+impl From<i32> for Fd {
+    fn from(fd: i32) -> Self {
+        Self(fd)
+    }
+}
+impl std::os::unix::io::AsRawFd for Fd {
+    fn as_raw_fd(&self) -> std::os::unix::io::RawFd {
+        self.0
+    }
+}
+impl std::ops::Deref for Fd {
+    type Target = i32;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+#[cfg_attr(feature = "unstable", doc(cfg(unix)))]
+#[derive(Debug)]
+#[allow(non_camel_case_types)]
+/// External memory that rely on unix file descriptors
+pub enum ExternalMemoryFd {
+    /// Tmp
+    OPAQUE_FD(Fd, u64),
```

Comment:
We should use the proper snake naming scheme

Reply:
I had chosen this name patter so that it was matching with the `ExternalMemoryTypeFlags`, but i had not thought that the user will anyway build that type with something "subtype".into(), so matching name was not so relevant. With the new code a lot of "subtypes" has removed and every enum is using snake naming scheme.